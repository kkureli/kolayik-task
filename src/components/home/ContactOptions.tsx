import React, {FC} from 'react';
import {Image, Linking, StyleSheet, TouchableOpacity, View} from 'react-native';
import {ContactMethods} from '../../models';
import Typography from '../common/Typography';

interface ContactProps {
  methods: ContactMethods[];
}

const ContactOptions: FC<ContactProps> = ({methods}) => {
  const onContact = (id: string) => {
    switch (id) {
      case 'email':
        console.log('Send Mail');
        //Linking.openURL(`mailto:${props.user.email}`)
        break;
      case 'call':
        console.log('Call');
        //Linking.openURL(`tel:${phoneNumber}`)
        break;
      default:
        break;
    }
  };

  return (
    <View style={styles.container}>
      {methods.map(method => {
        return (
          <TouchableOpacity
            onPress={() => onContact(method.id)}
            style={styles.option}
            key={method.id}>
            <Image style={styles.icon} source={method.icon} />
            <Typography subtitle medium center primary>
              {method.title}
            </Typography>
          </TouchableOpacity>
        );
      })}
    </View>
  );
};

export default ContactOptions;

const styles = StyleSheet.create({
  container: {
    flexDirection: 'row',
    justifyContent: 'space-around',
    marginTop: 30
  },
  icon: {width: 70, height: 70, resizeMode: 'contain', marginBottom: 10},
  option: {
    marginHorizontal: 5
  }
});
