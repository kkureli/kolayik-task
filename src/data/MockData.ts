import {ContactMethods, SubMenuModel, User} from '../models';
const avatar = require('../assets/images/Avatar.png');

const mockUser: User = {
  image: avatar,
  name: 'Adnan Burhan',
  title: 'Product Designer',
  email: 'adnan@burhan.com',
  phoneNumber: 5360700195,
  department: 'Product',
  location: 'Istanbul',
  employmentStartDate: '25 Jan 2019',
  manager: 'Yunus Türkoğlu'
};

const contactOptions: ContactMethods[] = [
  {
    title: 'Email',
    icon: require('../assets/icons/contactOptions/emailIcon.png'),
    id: 'email'
  },
  {
    title: 'Call',
    icon: require('../assets/icons/contactOptions/phoneCallIcon.png'),
    id: 'call'
  }
];

const subMenu: SubMenuModel[] = [
  {
    title: 'Requests',
    icon: require('../assets/icons/subMenuIcons/requestsIcon.png'),
    id: '1',
    onPress: () => console.log('navigation.navigate(RequestsScreen)')
  },
  {
    title: 'Information',
    icon: require('../assets/icons/subMenuIcons/infoIcon.png'),
    id: '2',
    onPress: () => console.log('navigation.navigate(InformationScreen)')
  },
  {
    title: 'Career',
    icon: require('../assets/icons/subMenuIcons/careerIcon.png'),
    id: '3',
    onPress: () => console.log('navigation.navigate(InformationScreen)')
  },
  {
    title: 'Leaves',
    icon: require('../assets/icons/subMenuIcons/leavesIcon.png'),
    id: '4',
    onPress: () => console.log('navigation.navigate(LeavesScreen)')
  },
  {
    title: 'Assets',
    icon: require('../assets/icons/subMenuIcons/assetsIcon.png'),
    id: '5',
    onPress: () => console.log('navigation.navigate(AssetsScreen)')
  },
  {
    title: 'Documents',
    icon: require('../assets/icons/subMenuIcons/documentsIcon.png'),
    id: '6',
    onPress: () => console.log('navigation.navigate(DocumentsScreen)')
  }
];

export {mockUser, subMenu, contactOptions};
