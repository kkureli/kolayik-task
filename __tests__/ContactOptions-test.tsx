import React from 'react';
import renderer from 'react-test-renderer';
import ContactOptions from '../src/components/home/ContactOptions';
import {contactOptions} from '../src/data/MockData';

test('contactOptions renders correctly', () => {
  const tree = renderer
    .create(<ContactOptions methods={contactOptions} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
