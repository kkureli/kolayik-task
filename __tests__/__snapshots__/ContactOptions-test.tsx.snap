// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`contactOptions renders correctly 1`] = `
<View
  style={
    Object {
      "flexDirection": "row",
      "justifyContent": "space-around",
      "marginTop": 30,
    }
  }
>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    nativeID="animatedComponent"
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "marginHorizontal": 5,
        "opacity": 1,
      }
    }
  >
    <Image
      source={
        Object {
          "testUri": "../../../src/assets/icons/contactOptions/emailIcon.png",
        }
      }
      style={
        Object {
          "height": 70,
          "marginBottom": 10,
          "resizeMode": "contain",
          "width": 70,
        }
      }
    />
    <Text
      margin={null}
      padding={null}
      style={
        Object {
          "color": "#376BFB",
          "fontFamily": "Roboto",
          "fontSize": 14,
          "fontWeight": "400",
          "textAlign": "center",
        }
      }
    >
      Email
    </Text>
  </View>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    nativeID="animatedComponent"
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "marginHorizontal": 5,
        "opacity": 1,
      }
    }
  >
    <Image
      source={
        Object {
          "testUri": "../../../src/assets/icons/contactOptions/phoneCallIcon.png",
        }
      }
      style={
        Object {
          "height": 70,
          "marginBottom": 10,
          "resizeMode": "contain",
          "width": 70,
        }
      }
    />
    <Text
      margin={null}
      padding={null}
      style={
        Object {
          "color": "#376BFB",
          "fontFamily": "Roboto",
          "fontSize": 14,
          "fontWeight": "400",
          "textAlign": "center",
        }
      }
    >
      Call
    </Text>
  </View>
</View>
`;
